src/compat/posix/proc/exec.c

```diff
@@ -69,10 +69,26 @@ int exec_call(void) {
 
 int execv(const char *path, char *const argv[]) {
 	struct task *task;
+	int i;
+	char cmd_name[MAX_TASK_NAME_LEN] = {0};
 	/* save starting arguments for the task */
 	task = task_self();
 	task_resource_exec(task, path, argv);
-	task_set_name(task, path);
+
+
+	for (i = 0; argv[i] != NULL; i ++) {
+		strncat(cmd_name, argv[i], MAX_TASK_NAME_LEN);
+		cmd_name[MAX_TASK_NAME_LEN - 1] = '\0';
+		strncat(cmd_name, " ", MAX_TASK_NAME_LEN);
+		cmd_name[MAX_TASK_NAME_LEN - 1] = '\0';
+
+		if (i >= 3){
+			// TODO for protection from a lot of arguments
```

Comment:
From man page
```
The list of arguments must be terminated by a null pointer
```

It's violation of exec interface, I suppose it could harm later too. Will be OK after comment change